release/release.go

```diff
@@ -536,25 +540,56 @@ func sqliteVersionBinding(sqliteVersion string) string {
 		word  = "SQLITE_VERSION"
 	)
 
-	submatch := findInURL(sqliteBindingURL, regex, word)
+	submatch := findInURL(sqliteBindingURL, regex, word, true)
 	if len(submatch) > 1 {
 		return submatch[1]
 	}
 
 	return ""
 }
 
+func createCalicoURL(calicoVersion string) string {
+	const (
+		regex    = `\"(.*)\"`
+		notFound = "Page Not Found"
+	)
+
+	var formattedVersion string
+
+	versionRegex := regexp.MustCompile(`^v(\d+\.\d+)(?:\.\d+)?$`)
+
+	formattedVersion = calicoVersion
+
+	// Check if the version matches the pattern
+	if versionRegex.MatchString(calicoVersion) {
+		matches := versionRegex.FindStringSubmatch(calicoVersion)
+		if len(matches) == 2 {
+			formattedVersion = "v" + matches[1]
+		}
+	}
+
+	calicoArchiveURL := "https://projectcalico.docs.tigera.io/archive/" + formattedVersion + "/release-notes/#" + strings.Trim(calicoVersion, "")
+
+	// check if doesn't exists content for archive url
+	submatch := findInURL(calicoArchiveURL, regex, notFound, false)
+	if len(submatch) > 1 {
+		return "https://docs.tigera.io/calico/latest/release-notes/#" + formattedVersion
+	}
+
+	return calicoArchiveURL
+}
+
 // findInURL will get and scan a url to find a slice submatch for all the words that matches a regex
 // if the regex is empty then it will return the lines in a file that matches the str
-func findInURL(url, regex, str string) []string {
+func findInURL(url, regex, str string, checkStatusCode bool) []string {
```

Comment:
It doesn't appear as though the bool is ever not true so I don't think it's necessary to have as an argument.